jwst/photom/photom.py

```diff
@@ -296,69 +296,34 @@ def calc_niriss(self, ftab):
             # data for each of the slits/orders in the input
             for slit in self.input.slits:
 
-                # Initialize the output conversion factor and increment slit number
-                match = False
+                # Increment slit number
                 self.slitnum += 1
 
                 # Get the spectral order number for this slit
                 order = slit.meta.wcsinfo.spectral_order
-
                 log.info("Working on slit: {} order: {}".format(slit.name, order))
 
-                # Locate matching row in reference file
-                for tabdata in ftab.phot_table:
-
-                    ref_filter = tabdata['filter'].strip().upper()
-                    ref_pupil = tabdata['pupil'].strip().upper()
-                    ref_order = tabdata['order']
-
-                    # Find matching values of FILTER, PUPIL, ORDER
-                    if (self.filter == ref_filter and self.pupil == ref_pupil and order == ref_order):
-                        self.photom_io(tabdata)
```

Comment:
@hbushouse The previous imlpementation did not pass order to  `photom_io` for Niriss and Nircam WFSS. This PR passes `order`. Let me know if this is not correct.

Reply:
I believe all of this is correct.  For NIRCam and NIRISS WFSS 2-D extracted spectra, the spectral order is included in the wcs model, so it is not passed as an argument when calling wcs; the spectral order number is saved in the metadata.  I don't recall seeing any mention of spectral order for NIRSpec fixed slits or MSA.  The `get_wavelengths` function in lib/wcs_utils.py has a spectral order argument, but that is only used for NIRISS SOSS, for which the `niriss_soss_set_input` function is called to set the spectral order.  `get_wavelengths` will return the wavelength array from the model, if that attribute is already populated (which it should be for 2-D extracted spectra).

Reply:
Thanks, removed `order`